core/invoker/src/main/scala/whisk/core/container/ContainerPool.scala

```diff
@@ -559,7 +560,7 @@ class ContainerPool(
             try {
                 info(this, s"making new container because none available")
                 startingCounter.next()
-                makeGeneralContainer(key, containerName, imageName, limits, action.exec.pull)
+                makeGeneralContainer(key, containerName, imageName, limits, action.exec.asInstanceOf[CodeExec[_]].pull)
```

Comment:
Note the `assume(action.exec.isInstanceOf[CodeExec[_]])` in Invoker.scala asserts this cast is ok.